lms/djangoapps/grades/new/course_grade.py

```diff
@@ -185,6 +187,19 @@ def compute_and_update(self, read_only=False):
             )
         )
 
+    def score_for_chapter(self, chapter_key):
```

Comment:
**Note to reviewers:** I extracted the additional grading computation code from the gating module and moved it here - in order to keep such logic within the grades layer.